### DISCUSSION TITLE
0.50.2

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#7701] [General]
## 0.50.2.0

* Added new logo for Windows Subsystem for Linux
* Enable hardware performance counters if the hardware supports them [GH 4678]
    A %USERPROFILE%\\.wslconfig option has been added to opt-out:
    ```
    [wsl2]
    hardwarePerformanceCounters=false
    ```
* Fix issue when printing system error messages that contain inserts.
* Update the user tile to launch in the user's home directory instead of C:\WINDOWS\System32
* Restore default signal dispositions for /etc/wsl.conf boot.command process to prevent zombie processes [GH 7575]
* Switch to using a static CRT for Windows binaries
* Use store API to download distributions via wsl.exe --install
* Add --no-launch option to wsl.exe --install
* Many updates to localized strings.
* Switch to an updated tar for importing / exporting WSL2 distributions.
* Update to the official 22000 sdk
* Strip Linux symbols for release builds
* Update Linux kernel to 5.10.74.3  
    * Update to upstream stable kernel release 5.10.74
    * Enable BPF Type Format (CONFIG_DEBUG_INFO_BTF) for use by eBPF tools [GH 7437]
    * Changed Dxgkrnl version to 2110
        * Implemented D3DKMTShareObjectWithHost
        * Fixed QueryStatistics VM bus alignment issue for the result
        * Implemented D3DKMTCreateSyncFile
        * Address upstream submission feedback
        * Moved d3dkmthk to include/uapi/misc
        * Replaces u32 by __u32 and u64 by __u64
        * Added "_" in front of the enumerator values to support including both WDK and Linux headers
        * Removed holes in the user mode visible structures to be compatible with 32 bit apps
        * Replaces pointer in the user mode visible structures with a define to be u64 for user mode apps
        * Fix build failure with GCC versions older than 8.1 [GH 7558]
    * Enable the Buffer Sharing and Sync File Frameworks (CONFIG_DMA_SHARED_BUFFER, CONFIG_SYNC_FILE) for Dxgkrnl usage

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/'>0.50.2</a>.</em>

**Comment by mavaddat** [comment]
To install, make sure to have [Windows Subsystem for Linux enabled](https://docs.microsoft.com/en-us/windows/wsl/install). As pointed out by @benhillis , since the `msix` is code signed by Microsoft, [developer mode](https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development) is not necessary. Then,
1. Download the `msixbundle` release file
2. Launch PowerShell as Administrator
3. Use cmdlet `Add-AppxPackage` with `ForceUpdateFromAnyVersion` and `ForceApplicationShutdown`:  
```pwsh
Add-AppxPackage -Path "$env:TEMP\MicrosoftEdgeDownloads\*\Microsoft.WSL_0.50.2.0_x64_ARM64.msixbundle" -ForceUpdateFromAnyVersion -ForceApplicationShutdown
```
## Warning

```diff  
- Step 3 will forcibly terminate any instances of WSL and/or Windows Store running in the background.
- To avoid disruptive loss of data, make sure that no critical processes are 
- running on the WSL backend when you run this command. 
```

**Reply by benhillis** [reply]
@mavaddat - the msix is store signed, do you still need developer mode enabled? You also only need the WSL optional component installed if you want WSL1 support.

**Reply by grgsim** [reply]
Can confirm I just updated successfully without dev mode enabled on Windows 11 22000.318

**Reply by mavaddat** [reply]
> Can confirm I just updated successfully without dev mode enabled on Windows [11 22000.318](\#11 22000.318)

Nice. I’ll modify my above advice accordingly.

**Comment by heartacker** [comment]
install this release will add another app or just update my installed wsl app? any body tell me. thanks

**Reply by EricCote** [reply]
It will update the installed wsl app.

**Comment by Biswa96** [comment]
> Added new logo for Windows Subsystem for Linux

Wish the new icons be released with a free and open source license.

**Comment by AlbanDAQUINO** [comment]
Installed via an Admin Powershell on Windows 11 21H2 (22000.346) Beta channel.

I just changed the path to my usual download folder and it worked.
Updated my WSL2 Debian 11 with the latest kernel. 👍🏼
I haven't checked my 7 other WSLs yet tho ... 🥲

Edit: Typos and ...
- Devuan 4 👍🏼
- Ubuntu 21.10 👍🏼
- Archlinux 👍🏼

**Reply by pnunn** [reply]
I know tihs is WAY off topic, but I tried the install and it seems I'm not on a recent enough version of windows.

Add-AppxPackage : Deployment failed with HRESULT: 0x80073CFD, A Prerequisite for an install could not be satisfied.
Windows cannot install package MicrosoftCorporationII.WindowsSubsystemforLinux_0.50.2.0_x64__8wekyb3d8bbwe because
this package is not compatible with the device. The package requires OS version 10.0.22000.0 or higher on the
Windows.Desktop device family. The device is currently running OS version 10.0.19043.1348.
NOTE: For additional information, look for [ActivityId] ba257b29-d9c6-0000-3a65-40bac6d9d701 in the Event Log or use
the command line Get-AppPackageLog -ActivityID ba257b29-d9c6-0000-3a65-40bac6d9d701
At line:1 char:1
+ Add-AppxPackage -Path "Microsoft.WSL_0.50.2.0_x64_ARM64.msixbundle" - ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (C:\Users\pjn_o\...RM64.msixbundle:String) [Add-AppxPackage], Exception
    + FullyQualifiedErrorId : DeploymentError,Microsoft.Windows.Appx.PackageManager.Commands.AddAppxPackageCommand

Which is actually a really useful error message for once.  How do I get the later version given I'm fully up to date (or do I have to join some beta program to get there?).

Thanks for your work and support.

**Reply by EricCote** [reply]
In the error message, OS version "10.0.22000" really means "Windows 11".

**Reply by pnunn** [reply]
Ah.. I see.. guess that will have to wait then. :(  Thanks for the info.

**Reply by Trass3r** [reply]
Thanks for trying it. I hope it will come to Win10 too some day.

**Comment by Trass3r** [comment]
>     * Enable hardware performance counters

Will this also enable tracing by accident?
`/sys/kernel/tracing` is mounted but empty no matter which tracer kernel configs I enable.

**Comment by yingdam** [comment]
> Enable hardware performance counters if the hardware supports them [GH 4678]

I wonder where can I find the GH 4678 and check whether my hardware is supported?

**Reply by Biswa96** [reply]
Here https://github.com/microsoft/WSL/issues/4678
